src/Core/Services/Cache/DabCacheService.cs

```diff
@@ -0,0 +1,137 @@
+    }
+
+    /// <summary>
+    /// Attempts to fetch response from cache. If there is a cache miss, call the 'factory method' to get a response
+    /// from the backing database.
+    /// </summary>
+    /// <typeparam name="JsonElement">Response payload</typeparam>
+    /// <param name="queryExecutor">Factory method. Only executed after a cache miss.</param>
+    /// <param name="queryMetadata">Metadata used to create a cache key or fetch a response from the database.</param>
+    /// <param name="cacheEntryTtl">Number of seconds the cache entry should be valid before eviction.</param>
+    /// <returns>JSON Response</returns>
+    /// <exception cref="Exception">Throws when the cache-miss factory method execution fails.</exception>
+    public async ValueTask<JsonElement?> GetOrSetAsync<JsonElement>(IQueryExecutor queryExecutor, DatabaseQueryMetadata queryMetadata, int cacheEntryTtl)
+    {
+        string cacheKey = CreateCacheKey(queryMetadata);
+        JsonElement? result = await _cache.GetOrSetAsync(
+               key: cacheKey,
+               async (FusionCacheFactoryExecutionContext<JsonElement> ctx, CancellationToken ct) =>
+               {
+                   // Need to handle undesirable results like db errors or null.
+                   JsonElement? result = await queryExecutor.ExecuteQueryAsync(
+                       sqltext: queryMetadata.QueryText,
+                       parameters: queryMetadata.QueryParameters,
+                       dataReaderHandler: queryExecutor.GetJsonResultAsync<JsonElement>,
+                       httpContext: _httpContextAccessor.HttpContext!,
+                       args: null,
+                       dataSourceName: queryMetadata.DataSource);
+
+                   ctx.Options.SetSize(EstimateCacheEntrySize(cacheKey: cacheKey, cacheValue: result?.ToString()));
+                   ctx.Options.SetDuration(duration: TimeSpan.FromSeconds(cacheEntryTtl));
+
+                   return result;
+               });
+
+        return result;
+    }
+
+    /// <summary>
+    /// Creates a cache key using the request metadata resolved from a built SqlQueryStructure
+    /// Format: DataSourceName:QueryText:JSON_QueryParameters
+    /// Example: 7a07f92a-1aa2-4e2a-81d6-b9af0a25bbb6:select * from MyTable where id = @param1 = :{"@param1":{"Value":"42","DbType":null}}
+    /// </summary>
+    /// <returns>Cache key string</returns>
+    private string CreateCacheKey(DatabaseQueryMetadata queryMetadata)
+    {
+        StringBuilder cacheKeyBuilder = new();
+        cacheKeyBuilder.Append(queryMetadata.DataSource);
+        cacheKeyBuilder.Append(KEY_DELIMITER);
+        cacheKeyBuilder.Append(queryMetadata.QueryText);
+        cacheKeyBuilder.Append(KEY_DELIMITER);
+        cacheKeyBuilder.Append(JsonSerializer.Serialize(queryMetadata.QueryParameters));
+        string cacheKey = cacheKeyBuilder.ToString();
+
+        if (_logger?.IsEnabled(LogLevel.Trace) ?? false)
```

Comment:
should we disable this logging in late configured scenario?

Reply:
We could, in that case we would need to evaluate logging across the dab solution to determine whether to log in late configured scenario wouldn't we? 

Are you highlighting a scenario specific to caching that should preclude logging in late bound config scenario?

Reply:
Removing for now, since cachekey will contain lots of request metadata that we may/may not want to log. Can revisit.